src/mlpack/methods/ann/layer/radial_basis_function_impl.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file radial_basis_function_impl.hpp
+ * @author Himanshu Pathak
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "radial_basis_function.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType,
+    typename Activation>
+RBF<InputDataType, OutputDataType, Activation>::RBF() :
+    inSize(0),
+    outSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename Activation>
+RBF<InputDataType, OutputDataType, Activation>::RBF(
+    const size_t inSize,
+    const size_t outSize,
+    arma::mat& centres) :
+    inSize(inSize),
+    outSize(outSize),
+    centres(centres)
+{
+  weights.set_size(outSize * inSize + outSize, 1);
+  sigmas = arma::ones(1, outSize);
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename Activation>
+template<typename eT>
+void RBF<InputDataType, OutputDataType, Activation>::Forward(
+    const arma::Mat<eT>& input, arma::Mat<eT>& output)
+{
+  sigmas = arma::ones(outSize, 1);
+  arma::cube x = arma::cube(input.n_rows, outSize, input.n_cols);
+
+  for (size_t i = 0; i < input.n_cols; i++)
+  {
+    x.slice(i).each_col() = input.col(i);
+  }
+
+  arma::cube c = arma::cube(input.n_rows, outSize, input.n_cols);
+  c.each_slice() = centres;
+
+  distances = arma::mat(input.n_rows, input.n_cols);
+
+  for (size_t i = 0; i < input.n_cols; i++)
+  {
+    distances.col(i) = arma::pow(arma::sum(
+                                 arma::pow((
+                                 x.slice(i) - c.slice(i)),
+                                 2), 1), 0.5);
+  }
+
+  Activation activation = Activation();
+  activation.Fn(distances, output);
```

Comment:
Do we need to create an instance here? can we do it how it is implemented in base layer?

Reply:
`ActivationFunction::Fn(distances, output)`